videostore/videostore.go

```diff
@@ -296,13 +296,20 @@ func (vs *videostore) Fetch(_ context.Context, r *FetchRequest) (*FetchResponse,
 		return nil, err
 	}
 	vs.logger.Debug("fetch command received")
-
 	fetchFilePath := generateOutputFilePath(
 		vs.config.Storage.OutputFileNamePrefix,
 		formatDateTimeToString(r.From),
 		"",
 		tempPath)
 
+	// Always attempt to remove the concat file after the operation.
+	// This handles error cases in Concat where it fails in the middle
+	// of writing.
+	defer func() {
+		if err := os.Remove(fetchFilePath); err != nil {
```

Comment:
check if the file exists with `os.Stat` if it doesn't early return, if it does, do the `os.Remove`

Reply:
Good call. Added stat lookup before Remove attempt.